librz/bin/dwarf.c

```diff
@@ -1220,9 +1220,11 @@ static int init_die(RzBinDwarfDie *die, ut64 abbr_code, ut64 attr_count) {
 	if (!die) {
 		return -1;
 	}
-	die->attr_values = calloc(sizeof(RzBinDwarfAttrValue), attr_count);
-	if (!die->attr_values) {
-		return -1;
+	if (attr_count) {
```

Comment:
Is this check really necessary? Isn't checking for `->capacity` in the various uses enough?
Also, this is `init_die` function, is it alright to have `die->attr_values` potentially not initialized? (i didn't check if the caller already initialize it).

Reply:
No, it is not strictly required. It only affects whether we then have `NULL` or whatever `calloc()` gives for a zero size. I kind of prefer the former.